python-sdk/src/astro/sql/operators/data_validations/ColumnCheckOperator.py

```diff
@@ -0,0 +1,242 @@
+            self.df = self.dataset
+        else:
+            raise ValueError("dataset can only be of type pandas.dataframe | Table object")
+
+        self.process_checks()
+
+    def get_check_result(self, check_name: str, column_name: str, df: pandas.DataFrame):
+        """
+        Get the check method results post validating the dataframe
+        """
+        if df is not None and column_name in df.columns:
+            column_checks = {
+                "null_check": self.col_null_check,
+                "distinct_check": self.col_distinct_check,
+                "unique_check": self.col_unique_check,
+                "min": self.col_min,
+                "max": self.col_max,
+            }
+            return column_checks[check_name](column_name=column_name, df=df)
+        elif df is None:
+            raise ValueError("Dataframe is None")
+        else:
+            raise ValueError(f"Dataframe is don't have column {column_name}")
+
+    def process_checks(self):
+        """
+        Process all the checks and print the result or raise an exception in the event of failed checks
+        """
+        failed_tests = []
+        passed_tests = []
+
+        # Iterating over columns
+        for column in self.column_mapping:
+            checks = self.column_mapping[column]
+
+            # Iterating over checks
+            for check in checks:
+                tolerance = self.column_mapping[column][check].get("tolerance")
+                result = self.get_check_result(check, column_name=column, df=self.df)
+                self.column_mapping[column][check]["result"] = result
+                self.column_mapping[column][check]["success"] = self._get_match(
+                    self.column_mapping[column][check], result, tolerance
+                )
+                failed_tests.extend(_get_failed_checks(self.column_mapping[column], column))
+                passed_tests.extend(_get_success_checks(self.column_mapping[column], column))
+
+        if len(failed_tests) > 0:
+            raise AirflowException(f"The following tests have failed:" f"\n{''.join(failed_tests)}")
+        if len(passed_tests) > 0:
+            print(f"The following tests have passed:" f"\n{''.join(passed_tests)}")
+
+    @staticmethod
+    def col_null_check(column_name: str, df: pandas.DataFrame) -> Optional[int]:
+        """
+        Count the total null values in a dataframe column
+        """
+        return list(df[column_name].isnull().values).count(True)
+
+    @staticmethod
+    def col_distinct_check(column_name: str, df: pandas.DataFrame) -> Optional[int]:
+        """
+        Count the distinct value in a dataframe column
+        """
+        return len(df[column_name].unique())
+
+    @staticmethod
+    def col_unique_check(column_name: str, df: pandas.DataFrame) -> Optional[int]:
+        """
+        Count the unique value in a dataframe column
+        """
+        return len(df[column_name]) - len(df[column_name].unique())
```

Comment:
Aren't these the non unique ones?

Reply:
```suggestion
    def col_non_unique_check(column_name: str, df: pandas.DataFrame) -> Optional[int]:
        """
        Count the non-unique values in a dataframe column
        """
        return df[column_name].count()
```
https://pandas.pydata.org/docs/reference/api/pandas.Series.count.html

Reply:
@feluelle I'm not entirely sure of the intent here, here I'm just trying to mimic the condition in the parent operator. 
SQL - "unique_check": "COUNT({column}) - COUNT(DISTINCT({column}))"